feat: Load CLR with Python.NET

LeeDongGeon1996 opened the pull request:
#### changes
 - Load CLR with Python.NET

- [ ] Implement
- [x] Fix: #420

LeeDongGeon1996 commented:
both failed actions look like authorization issue.

DaveSkender commented:
> both failed actions look like authorization issue.

- [ ] did a quick fix in #422 for Actions issue
- [ ] I’ll leave some convo notes in #420 for the underlying problem
- [ ] will also review this after convo

DaveSkender commented on stock_indicators/_cslib/runtimeconfig.json:
I’d assumed this might help “hint” the Python.NET CLR loader with compatibility information about the DLL, so reconsider if useful.  Just an FYI, the `rollforward` aspect of this effectively says the env should _at least have_ .NET 6 or some form of backwards compatibility.  So, just having any newer on that can interpret v6 will work.

LeeDongGeon1996 replied:
I'm not sure,, but can we get additional descriptive message from that when failed to load clr? The requirement for `runtimeconfig.json` is removed, it is optional now. so I think we don't need more hint for it. Please let me know if there's other useful feature :)

LeeDongGeon1996 replied:
https://pythonnet.github.io/pythonnet/python.html#calling-pythonnet-load

DaveSkender commented on stock_indicators/_cslib/__init__.py:
```diff
@@ -24,52 +21,16 @@
-
-
 try:
-    verify_dotnet_environment()
```
no issue with removing this in favor of the built-in `pythonnet` [find.py](https://github.com/pythonnet/clr-loader/blob/master/clr_loader/util/find.py#L19-L59) utility, if it works, but you may have to explicitly call it here.

LeeDongGeon1996 replied:
@DaveSkender Built-in utility(`find_dotnet_root()`) is called inside `load()`. we can call explicitly but it might be redundant.

DaveSkender reviewed (commented):
Check unit tests as I’ve now added an ARM64 runner for checks that are currently failing.  It may be, that this more ideal Python.NET `find_dotnet_root()` utility isn’t actually working yet, for this situation.

LeeDongGeon1996 commented:
@DaveSkender Looks like the failed tests were false negatives for some reason. I re-run all tests and it all passed. Does it all  passed before this PR?

LeeDongGeon1996 commented:
The root problem was my virtual environment on my local machine is on M1 Rosetta mode though. This proves the need that we should load clr through `pythonnet`. `pythonnet` is handling this case also.

DaveSkender commented:
> The root problem was my virtual environment on my local machine is on M1 Rosetta mode though. This proves the need that we should load clr through `pythonnet`. `pythonnet` is handling this case also.

I'm wondering if the root problem from #408 was actually with `venv`?  We might need a build test for that.  I removing my own local `venv` instance a couple weeks ago due to problems.

DaveSkender reviewed (commented):
Just re-request a Review when you want me to look at this again.  I'm not sure if you'd intended to or are interested in checking out these two open topics:

1. Typings issue with `clr` .NET members, even if just dev env problem?
2. Broken `venv` handling, or if that's even related to our code.

LeeDongGeon1996 commented:
> I'm wondering if the root problem from https://github.com/facioquo/stock-indicators-python/issues/408 was actually with venv? We might need a build test for that. I removing my own local venv instance a couple weeks ago due to problems.

@DaveSkender  Sorry, I made you confused.

The reason for #408:
 -  This is because `AddReference()` does not work correctly on .NET9 over MacOS. [You added DLL manually and now it works](https://github.com/facioquo/stock-indicators-python/blob/main/stock_indicators/_cslib/__init__.py#L81-L97). I think finding `DOTNET_ROOT` manually is not a solution for here. That's why I removed that lines in this PR.

The reason for error on my virtual env:
 - Because my virtual env was on `M1 Rosetta mode`, DotNet SDK path must be `/usr/local/share/dotnet/x64`, which is ended with `x64`. But our [main branch currently is now fixed to `/usr/local/share/dotnet` explicitly](https://github.com/facioquo/stock-indicators-python/blob/main/stock_indicators/_cslib/__init__.py#L35). It caused the CLR load error on my machine.

LeeDongGeon1996 commented:
I don't think we need a bulid test for `venv`. This should be on `pythonnet` their side. and it is already covered by them.

LeeDongGeon1996 commented:
> Typings issue with clr .NET members, even if just dev env problem?

@DaveSkender 
I'm not sure but as I know that linting for .NET members are not supported since they are not python native things or even they don't provide the interface(like `.pyi`) that linter can recognize.

DaveSkender reviewed (commented):
Makes sense, intuitively.  One thing I'll want to check is whether macOS with just .NET 9 still works.  The GitHub runners have .NET 8 and .NET 9 installed, so we many not be getting an entirely pure test on that scenario.

In my head, I'm still confused by some of this -- if we rollback the changes I made to fix a problem; how is it that the problem is still solved?

DaveSkender reviewed (approved):
I think I understand this better now, and was able to run in on a macOS with just .NET 9 installed.
